engine/src/main/java/io/zeebe/engine/processor/TypedRecord.java

```diff
@@ -22,8 +22,12 @@
 
 public interface TypedRecord<T extends UnpackedObject> {
 
+  long getPosition();
```

Comment:
I don't think this method should be in the interface. It is not always implement, i.e. `TypedWorkflowInstanceRecord` always returns 0. (https://github.com/zeebe-io/zeebe/pull/2491/files#diff-37ffb731f1b5ec8f32004d4b5ca882ecR51)

Also it is not used in any `TypedRecordProcessor` implementation. It is only used inside the `ProcessingStateMachine` where we have access to the `TypedEventImpl` which contains the position.